backend/src/utils/adminUtils.ts

```diff
@@ -0,0 +1,151 @@
+import { CustomObjectsApi } from '@kubernetes/client-node';
+import { KubeFastifyInstance } from '../types';
+import { getAdminGroups, getAllowedGroups, getGroup } from './groupsUtils';
+
+const SYSTEM_AUTHENTICATED = 'system:authenticated';
+
+export const getGroupsConfig = async (
+  fastify: KubeFastifyInstance,
+  customObjectApi: CustomObjectsApi,
+  username: string,
+): Promise<boolean> => {
+  try {
+    const adminGroups = getAdminGroups();
+    const adminGroupsList = adminGroups.split(',');
+
+    if (adminGroupsList.includes(SYSTEM_AUTHENTICATED)) {
+      throw new Error('It is not allowed to set system:authenticated or as admin group.');
```

Comment:
oh, no no, it¡s not possible to add system:authenticated trough the UI, although since this is a non-authenticated request to the dashboard someone could spoof the body, so that's why I added this failsafe in backend, I tend to add this kind of checks in backend even though the logic in the frontend doesn't allow it just to avoid bugs.